src/netCDF4/__init__.pyi

```diff
@@ -541,14 +541,14 @@ class Variable(Generic[VarT]):
         datatype: DatatypeSpecifier,
         dimensions: DimensionsSpecifier = (),
         compression: CompressionType | str | None = None,
-        zlib: Any = False,
+        zlib: bool = False,
         complevel: CompressionLevel | int | None = 4,
-        shuffle: Any = True,
-        szip_coding: Literal["nn", "ec"] = "nn",
-        szip_pixels_per_block: Literal[4, 8, 16, 32] = 8,
-        blosc_shuffle: Literal[0, 1, 2] = 1,
-        fletcher32: Any = False,
-        contiguous: Any = False,
+        shuffle: bool = True,
+        szip_coding: Literal["nn", "ec"] | str = "nn",
```

Comment:
Do you think that it occurs often that people somehow pass this dynamically instead of directly in the `__init__`?
I know that it probably happens in tests, but in real-world scenarios I doubt that leaving it only as literal will cause problems.

Reply:
It's precisely errors I got in the tests that caused me to make those changes. I don't know what the right answer is here. If you think it's better to only allow literals in the stubs I could be convinced.

Reply:
looking at other projects I see that the str union is only added if the literals are dynamically extended by further projects or are simply too many. If it is not too much work, I would prefer to keep the literal only (But no strong preference).